docs/services.md

```diff
@@ -208,66 +264,104 @@ DNS records for each.  If DNS has been enabled throughout the cluster then all
-is that all such `Service`s share a port space - only one of them can have port
-80, for example.
+address.  Kubernetes supports two ways of doing this: `NodePort`s and
+`LoadBalancer`s.
 
-## Choosing your own PortalIP address
+Every `Service` has a `Type` field which defines how the `Service` can be
+accessed.  Valid values for this field are:
+   - ClusterIP: use a cluster-internal VIP (portal) only - this is the default
+   - NodePort: use a cluster IP, but also expose the service on a port on each
+     node of the cluster (the same port on each)
+   - LoadBalancer: use a ClusterIP and a NodePort, but also ask the cloud
+     provider for a load balancer which forwards to the `Service`
 
-A user can specify their own `PortalIP` address as part of a service creation
-request.  For example, if they already have an existing DNS entry that they
-wish to replace, or legacy systems that are configured for a specific IP
-address and difficult to re-configure.  The `PortalIP` address that a user
-chooses must be a valid IP address and within the portal net CIDR range that is
-specified by flag to the API server.  If the PortalIP value is invalid, the
-apiserver returns a 422 HTTP status code to indicate that the value is invalid.
+Note that while `NodePort`s can be TCP or UDP, `LoadBalancer`s only support TCP
+as of Kubernetes 1.0.
+
+### Type = NodePort
+
+If you set the `type` field to `"NodePort"`, the Kubernetes master will
+allocate you a port (from a flag-configured range) on each node for each port
+exposed by your `Service`.  That port will be reported in your `Service`'s
+`spec.ports[*].nodePort` field.  If you specify a value in that field, the
+system will allocate you that port or else will fail the API transaction.
+
+This gives developers the freedom to set up their own load balancers, to
+configure cloud environments that are not fully supported by Kubernetes, or
+even to just expose one or more nodes' IPs directly.
+
+### Type = LoadBalancer
+
+On cloud providers which support external load balancers, setting the `type`
+field to `"LoadBalancer"` will provision a load balancer for your `Service`.
+The actual creation of the load balancer happens asynchronously, and
+information about the provisioned balancer will be published in the `Service`'s
+`status.loadBalancer` field.  For example:
+
+```json
+{
+    "kind": "Service",
+    "apiVersion": "v1beta3",
+    "metadata": {
+        "name": "my-service",
+    },
+    "spec": {
+        "selector": {
+            "app": "MyApp"
+        },
+        "ports": [
+            {
+                "protocol": "TCP",
+                "port": 80,
+                "targetPort": 9376,
+                "nodePort": 30061
+            }
+        ],
+        "portalIP": "10.0.171.239",
+        "type": "NodePort"
+    },
+    "status": {
+        "loadBalancer": {
+            "ingress": [
+                {
+                    "ip": "146.148.47.155"
```

Comment:
I mentioned it, I don't feel like we need to go into too much depth here right now.